extension/src/state_aggregate/rollup.rs

```diff
@@ -142,22 +146,36 @@ impl OwnedCompactStateAgg {
 
         let gap = later.first_time - earlier.last_time;
         assert!(gap >= 0);
-        merged_durations[earlier.last_state as usize].duration += gap;
+        merged_durations
+            .get_mut(earlier.last_state as usize)
+            .expect("earlier.last_state doesn't point to a state")
```

Comment:
Are these error messages meant to be for us or for end users?  As written, these will be very helpful in letting us see exactly went wrong, but any cloud user who gets an error like this is likely going to have a hard time making any sense of it.

Reply:
That error message should never be displayed to an end user — it would only happen if the aggregate is already in an inconsistent state. If a user hits this error it's because of a bug somewhere.